core/psd/imx6ull/bcb.c

```diff
@@ -63,126 +63,141 @@ int dbbt_block_is_bad(dbbt_t *dbbt, uint32_t block_num)
 }
 
 
-int dbbt_flash(oid_t oid, FILE *f, dbbt_t *dbbt)
+int dbbt_flash(oid_t oid, int fd, dbbt_t *dbbt, const flashsrv_info_t *info)
 {
 	int i, err;
 	void *data;
-	offs_t partoff = 0, partsz;
 
-	if ((err = flashmng_getAttr(atSize, &partsz, oid)) < 0)
-		return err;
+	/* DBBT is just after BCB_CNT FCB blocks on the same partition */
+	offs_t partoff = BCB_CNT * info->erasesz;
 
-	if (partsz < (BCB_CNT * PAGES_PER_BLOCK * FLASH_PAGE_SIZE))
+	data = malloc(info->writesz);
+	if (!data)
 		return -1;
 
-	data = mmap(NULL, FLASH_PAGE_SIZE, PROT_READ | PROT_WRITE, MAP_UNCACHED, OID_NULL, 0);
-
 	dbbt_fingerprint(dbbt);
 	dbbt->size = 1;
 
 	for (i = 0; i < BCB_CNT; i++) {
-		if ((err = fseek(f, partoff, SEEK_SET)) < 0)
+		if ((err = lseek(fd, partoff, SEEK_SET)) < 0) {
+			printf("failed to lseek(1) to 0x%llx\n", partoff);
 			return err;
```

Comment:
Memleak here and below, I assume that we don't care.

Reply:
don't really care but added `free` for it to look nice